### ISSUE TITLE
Support `portalocker` >=2

### ISSUE THREAD

**Issue by dhirschfeld** [AzureAD/microsoft-authentication-extensions-for-python#94]
The current constraints restrict `portalocker` to <2:
https://github.com/AzureAD/microsoft-authentication-extensions-for-python/blob/a0cfe93433400b2d4ea673ebdef4bf064243c13d/setup.py#L25-L26

IIUC the major version bump was just to remove py2 support so this library *should* be compatible with `portalocker>=1.6`

**Comment by dhirschfeld** [comment]
***xref:*** https://portalocker.readthedocs.io/en/latest/#python-2

**Comment by rayluo** [comment]
Thanks, Dave, for bringing this to our attention.

Hi Rick (@wolph), would you please confirm whether Portalocker 2.x contains breaking change in its API when comparing to 1.x? The [2.0.0 release log](https://github.com/WoLpH/portalocker/releases/tag/v2.0.0) mentioned the removal of Python 2 support. We wonder whether our usage of Portalocker 1.x API would continue to work with Portalocker 2.x, when running on Python 3.5+.

**Comment by WoLpH** [comment]
Hi Ray,

The major differences between Portalocker 2.x and 1.x are currently:

1. Full type hinting support
2. Related to the type hinting, there is now a `LockFlags` enum that can be used: https://portalocker.readthedocs.io/en/latest/portalocker.constants.html?highlight=lockflags#portalocker.constants.LockFlags
3. Support for locks across multiple machines using Redis

For all intents and purposes all Portalocker 1.x code still works on Portalocker 2.x and will continue to do so.

Portalocker 1.x (and Python 2 support) is considered end-of-life but there are no known issues currently. And to maintain both forwards and backwards compatibility I could even backport the `LockFlags` if desired.

Long story short: yes, you can safely upgrade :)
